### ISSUE TITLE
Swap event data

### ISSUE THREAD

**Issue by cpvalente** [cpvalente/ontime#403]
Given a rundown with two events, we are currently able to swap the position of the event using the drag-and-drop feature.
However, there is no way to swap two events, while keeping the schedule. ie:

```
Event A > 09:00 - 09:30
Event B > 09:30 - 10:00
Event C > 10:00 - 10:30
```

If Event B were to run late, we would like to swap its place with Event C, keeping the schedule unchanged so that the rundown would read:
```
Event A > 09:00 - 09:30
Event C > 09:30 - 10:00
Event B > 10:00 - 10:30
```

### Requirements
- The user should be able to use the context menu to swap the data between two <EventBlocks>
- The timer (start, end, duration) data should remain as before, and only the remaining information should be swapped
- We would create a REST endpoint specifically for this action

### UI UX guidelines
We do not have a clear picture in UI as of now. Few guidelines:

On right-clicking an event block, we would have a new action: "Add to swap", which would add the `eventId` to the state of the context
If we right click an EventBlock, and we have a previously added an `eventId`, we would now see an added action `Swap with $eventId`

This would trigger the new endpoint created with both event Ids (to and from)

**Comment by cpvalente** [comment]
@asharonbaltazar 
I have made some revisions here, hoping to get your feedback. 

We would like our `ContextMenuContext` to remain generic, so it would not be ideal to create implementation-specific logic in there.
I wonder if a good way forwards is to create a small Zustand store that uses the `ContextMenuContext`

**Comment by asharonbaltazar** [comment]
If you are referring to the feature where a user can select events to swap using the context menu and ontime needs a way to track the selections, I think a Zustand store makes sense. Also, if I had known you were using Zustand, I'd probably written `ContextMenuContext` to use it as well.

**Comment by asharonbaltazar** [comment]
> If Event B were to run late, we would like to swap its place with Event C, keeping the schedule unchanged so that the rundown would read:
```ts
Event A > 09:00 - 09:30
Event C > 09:30 - 10:00
Event A > 10:00 - 10:30
```

Is there a typo here? Did you mean to write:
```ts
Event A > 09:00 - 09:30
Event C > 09:30 - 10:00
Event B > 10:00 - 10:30
```

Also, what did you mean by this?
> The user should be able to use drag-and-drop to swap the data between two 

Did you mean the present DND functionality should remain intact, and this new feature should be added to the context menu? Or did you mean changing the current DND functionality to swap the time between the event blocks and add the same functionality to the context menu?

**Comment by cpvalente** [comment]
> > If Event B were to run late, we would like to swap its place with Event C, keeping the schedule unchanged so that the rundown would read:
> 
> ```ts
> Event A > 09:00 - 09:30
> Event C > 09:30 - 10:00
> Event A > 10:00 - 10:30
> ```
> 
> Is there a typo here? Did you mean to write:
> 
> ```ts
> Event A > 09:00 - 09:30
> Event C > 09:30 - 10:00
> Event B > 10:00 - 10:30
> ```
> 
> Also, what did you mean by this?
> 
> > The user should be able to use drag-and-drop to swap the data between two
> 
> Did you mean the present DND functionality should remain intact, and this new feature should be added to the context menu? Or did you mean changing the current DND functionality to swap the time between the event blocks and add the same functionality to the context menu?

Thank you for your questions.
The first one, it was a mistake from my part, I have corrected the example

As for the second part, I was attempting at offering a choice to the developer implementing, we could either do this swap data as an extension of the current DND or as a context menu action.
I believe the latter is cheaper and potentially better. So I have also edited the text for clarification

In short, you had it right:
- current dnd remains intact, new functionality is implemented in the context menu

**Comment by asharonbaltazar** [comment]
I think the context menu is more suited since DND is reserved for another operation. 
Thanks for clarifying!

**Comment by asharonbaltazar** [comment]
https://github.com/cpvalente/ontime/assets/58940073/e59e3ce1-8ef9-4d32-8be4-9d39f478a438

Before I start working on the server side of things, this is the behavior you want, correct?

**Comment by cpvalente** [comment]
> Screen.Recording.2023-07-26.at.11.51.38.AM.mov
> 
> Before I start working on the server side of things, this is the behavior you want, correct?

This looks good to me.
Would you mind removing the extraneous curly brace in the ID while you are in the vicinity?
